### PR TITLE
Add GA4 tracking information

### PR DESCRIPTION
**Pull request by gtback**
Once the preview build finishes, this can be tested at https://docs_2525.docs-preview.app.elstc.co/guide/index.html

### CODE REVIEW

File: resources/web/docs_js/index.js [hunk 1]
```diff
@@ -379,6 +405,30 @@ $(function() {
     }, 150);
   }
 
+  window.dataLayer = window.dataLayer || [];
+
+  let titleParams = document.title.split('|')
+
+  let pageViewData = {
+    'event': 'page_view',
+    'pagePath': window.location.pathname,
```
Comment (gtback) [inline resources/web/docs_js/index.js:414]: Should these be uppercase `PagePath` or lower case `pagePath`. I'm seeing differences between the specification and the examples on www.elastic.co.

File: resources/web/docs_js/index.js [hunk 2]
```diff
@@ -379,6 +405,30 @@ $(function() {
...
+    'pageCategory': titleParams[titleParams.length - 2].trim(),
+    'hostname': window.location.hostname,
+    'canonicalTag': window.location.protocol + '//' + window.location.hostname + window.location.pathname,
+    'euid': getEuid(),
+    // 'userId': '', // TBD
+    // 'hashedIP': '', // TBD
+    'userAgent': window.navigator.userAgent,
+    ...getUtm()
+  };
+  console.log(pageViewData);
```
Comment (gtback) [inline resources/web/docs_js/index.js:428]: TODO: Remove before merging.

File: resources/web/docs_js/index.js [hunk 3]
```diff
@@ -211,6 +211,32 @@ function highlight_otp() {
   })
 }
 
+function getUtm() {
+  let qs = new URLSearchParams(window.location.search);
+  return {
+    'utm_source': qs['utm_source'],
```
[the remark this answers is not in the document]
Comment (gtback) [reply resources/web/docs_js/index.js:220]: Thanks, @brianjolly . Should be fixed in 5f414e901584295ef689aa0883ba157cfede5e9b. I don't know why it didn't work (I tried to simplify some other code, but apparently didn't do it correctly; this is why you shouldn't have me writing JavaScript 😉 ).

[the remark this answers is not in the document]
Comment (gtback) [reply resources/web/docs_js/index.js:220]: It will take ~30 minutes for a new build to be live.

**Review by brianjolly — APPROVED** [review]
This looks good to me!

File: resources/web/docs_js/index.js (hunk 2 above)
Comment (gtback) [reply resources/web/docs_js/index.js]: Fixed in d59e0cc7071ebacf566cfb5b9b361c4cd7e7df41

**Pull request merged**